docs/registry.md

```diff
@@ -0,0 +1,274 @@
+# Registry
+
+The registry is a place where applications metadata are stored and versioned. It can be used by a cozy 
+
+We define an application registry as an API. This should allow us to defer the real implementation of the registry storage and allow different store implementations.
+
+
+## Objects
+
+Two type of objects are managed in the registry: applications and versions.
+
+### Application
+
+An application object contains the following fields:
+
+- name: the application name
+- editor: the application editor
+- category: the application category
+- repository: object with type and url of package repository
+- license: the application license
+- versions: a list of all versions
+
+### Version
+
+An application version object contains the following fields:
+
+- name: the application name
+- description: description from the package.json
+- version: the version string
+- url: url of the tarball containing the application at specified version
+- sha256: the sha256 checksum of the application content
+- permissions: the permissions map contained in the manifest
+
+## APIs: Adding to registry
+
+These APIs can be used to add elements to the registry.
+
+### POST /:app
```

Comment:
For simplicity's sake, we could only have the add version route, with app being automatically created when the first version is uploaded.